packages/router/utils.ts

```diff
@@ -930,7 +928,7 @@ export function matchPath<
       if (isOptional && !value) {
         memo[paramName] = undefined;
       } else {
-        memo[paramName] = safelyDecodeURIComponent(value || "", paramName);
+        memo[paramName] = (value || "").replace(/%2F/g, "/");
```

Comment:
We don't need to call `decodeURIComponent` on param values here anymore since we matched against a slash-delimited path of segments that already went through `decodeURIComponent`